### PR TITLE
Fixed bug #574: SCA do not follow-up when creating an issue

### PR DESCRIPTION
**Pull request by NimrodGolan**
### Description

> Fixed bug #574: SCA do not follow-up when creating an issue

### References

> #574 

### Testing
TBD

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [X] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 1469 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**56 Medium**<br />Low&nbsp;**1314 Low**<br />Info&nbsp;**99 Info**<br />
### Violation Summary
Medium&nbsp;**26** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[282](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/AbstractVulnerabilityScanner.java#L282) [285](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/AbstractVulnerabilityScanner.java#L285) [299](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/AbstractVulnerabilityScanner.java#L299) |Medium|Use_of_System_exit|src/main/java/com/checkmarx/flow/service/AbstractVulnerabilityScanner.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=51)
[90](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/SastScanner.java#L90) [126](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/SastScanner.java#L126) |Medium|Use_of_System_exit|src/main/java/com/checkmarx/flow/service/SastScanner.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=55)
[142](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L142) [189](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L189) [210](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L210) [255](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L255) [269](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L269) [283](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L283) [290](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L290) [440](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/CxFlowRunner.java#L440) |Medium|Use_of_System_exit|src/main/java/com/checkmarx/flow/CxFlowRunner.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=43)
[54](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/OsaScannerService.java#L54) |Medium|Use_of_System_exit|src/main/java/com/checkmarx/flow/service/OsaScannerService.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=54)
[100](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/GitHubController.java#L100) [241](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/GitHubController.java#L241) |Medium|Unchecked_Input_for_Loop_Condition|src/main/java/com/checkmarx/flow/controller/GitHubController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=7)
[53](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/BitbucketCloudController.java#L53) [143](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/BitbucketCloudController.java#L143) |Medium|Unchecked_Input_for_Loop_Condition|src/main/java/com/checkmarx/flow/controller/BitbucketCloudController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=12)
[213](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/utils/ScanUtils.java#L213) |Medium|Unchecked_Input_for_Loop_Condition|src/main/java/com/checkmarx/flow/utils/ScanUtils.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=9)
[253](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/BitbucketServerController.java#L253) |Medium|Unchecked_Input_for_Loop_Condition|src/main/java/com/checkmarx/flow/controller/BitbucketServerController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=11)
[66](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/FlowController.java#L66) [69](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/FlowController.java#L69) [70](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/FlowController.java#L70) [77](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/FlowController.java#L77) [121](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/controller/FlowController.java#L121) |Medium|Unchecked_Input_for_Loop_Condition|src/main/java/com/checkmarx/flow/controller/FlowController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=3)
[790](https://github.com/checkmarx-ltd/cx-flow/blob/pr-nimrod-574/src/main/java/com/checkmarx/flow/service/JiraService.java#L790) |Medium|Privacy_Violation|src/main/java/com/checkmarx/flow/service/JiraService.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1038550&projectid=677&pathid=61)

File: src/main/java/com/checkmarx/flow/custom/ADOIssueTracker.java [hunk 1]
```diff
@@ -235,7 +235,7 @@ else if(!ScanUtils.empty(request.getApplication())){
         CreateWorkItemAttr title = new CreateWorkItemAttr();
         title.setOp("add");
         title.setPath(Constants.ADO_FIELD.concat(TITLE_FIELD));
-        title.setValue(getXIssueKey(resultIssue, request));
+        title.setValue(request.getProduct().getProduct() + " " + getXIssueKey(resultIssue, request));
```
Comment (alex-ko-dev) [inline src/main/java/com/checkmarx/flow/custom/ADOIssueTracker.java:238]:
```
request.getProduct().getProduct() + " " + getXIssueKey(resultIssue, request)
```
This snippet now appears 5 times throughout the code. I suggest that we put this logic into a single place (HTMLHelper?)

Comment (NimrodGolan) [reply src/main/java/com/checkmarx/flow/custom/ADOIssueTracker.java]: Done.

**Review by alex-ko-dev — APPROVED** [review]

**Pull request merged**